src/liballoc/collections/btree/set.rs

```diff
@@ -109,6 +109,80 @@ pub struct Range<'a, T: 'a> {
     iter: btree_map::Range<'a, T, ()>,
 }
 
+/// Core of an aLternative to Peekable, that is more efficient
+/// if copying items is cheap and if peeking early does no harm.
+/// It always has one item read ahead.
+#[derive(Clone)]
+struct Peeking<I>
+where
+    I: Iterator,
+    I::Item: Copy,
+{
+    head: Option<I::Item>,
```

Comment:
```suggestion
    /// Head is always the next element in the sequence, or None when empty
    head: Option<I::Item>,
```
I guess `head` is a great name, but a comment to explain that head is always filled (and not just on demand) would be good

Reply:
Uh, this is also in the doc just above, so.. yeah forget my suggestion :)